pac4j-core/src/main/java/org/pac4j/core/io/InputStreamSource.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2002-2012 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * 
+ *  This is a copy of org.springframework.core.io.InputStreamResource
+ */
+package org.pac4j.core.io;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**

```

Comment:
As we discussed I need a spring type abstraction on the file stream so I can store the file in a backend rather than on the file system. As I understood you didn't want a spring dependency to make this happen so I copied in the relavent files as is.. Except for FilesystemResource since I didn't need the full implementation hence the TODOs..